substrate/zombienet-sdk/tests/staking_next/tests.rs

```diff
@@ -0,0 +1,380 @@
+				break
+			}
+		}
+	}
+
+	Ok(())
+}
+
+// TODO: There should be a better way to do this.
+fn parse_paged_election_proceeded(fields: Composite<u32>) -> Option<RcClientEvent> {
+	if let subxt::ext::scale_value::Composite::Named(fields) = fields {
+		assert!(fields.len() == 2, "Expected 2 fields");
+		assert!(fields[0].0 == "page", "Expected page");
+
+		let page_idx = fields[0].1.as_u128().unwrap();
+
+		if let ValueDef::Variant(result) = &fields[1].1.value {
+			if result.name == "Err" {
+				return Some(RcClientEvent::PagedElectionProceeded {
+					page_idx,
+					page_content: Err(()),
+				})
+			}
+
+			assert!(result.name == "Ok", "Expected Ok");
+
+			if let subxt::ext::scale_value::Composite::Unnamed(r) = &result.values {
+				return Some(RcClientEvent::PagedElectionProceeded {
+					page_idx,
+					page_content: Ok(r[0].as_u128().unwrap()),
+				})
+			} else {
+				assert!(false, "Expected unnamed composite");
+			}
+		} else {
+			assert!(false, "Expected variant");
+		}
+	} else {
+		assert!(false, "Expected named composite");
+	}
+
+	None
+}
+
+fn parse_new_validator_set_count(fields: Composite<u32>) -> Option<AhClientEvent> {
+	if let subxt::ext::scale_value::Composite::Named(fields) = fields {
+		assert!(fields.len() == 4, "Expected 4 fields");
+
+		// we are interested only in `new_validator_set_count`
+
+		assert!(fields[1].0 == "new_validator_set_count", "Expected new_validator_set_count");
+
+		let new_validator_set_count = fields[1].1.as_u128().unwrap();
+
+		return Some(AhClientEvent::NewValidatorSetCount(new_validator_set_count))
+	}
+
+	None
+}
+
+fn parse_session_report_received(fields: Composite<u32>) -> Option<RcClientEvent> {
+	if let subxt::ext::scale_value::Composite::Named(fields) = fields {
+		assert!(fields.len() == 4, "Expected 4 fields");
+		assert!(fields[1].0 == "activation_timestamp", "Expected activation_timestamp");
+
+		if let ValueDef::Variant(activation_timestamp_variant) = &fields[1].1.value {
+			if activation_timestamp_variant.name == "None" {
+				return Some(RcClientEvent::SessionReportReceived { activation_time_stamp: None })
+			}
+
+			assert!(activation_timestamp_variant.name == "Some", "Expected Some");
+
+			if let subxt::ext::scale_value::Composite::Unnamed(act) =
+				&activation_timestamp_variant.values
+			{
+				if let ValueDef::Composite(a) = &act[0].value {
+					if let subxt::ext::scale_value::Composite::Unnamed(a) = &a {
+						return Some(RcClientEvent::SessionReportReceived {
+							activation_time_stamp: Some(a[0].as_u128().unwrap()),
+						})
+					} else {
+						log::info!("act: {:?}", act);
+						assert!(false, "Expected ValueDef::Primitive`(");
+					}
+				} else {
+					log::info!("act: {:?}", act);
+					assert!(false, "Expected ValueDef::Composite(");
+				}
+			} else {
+				assert!(false, "Expected unnamed composite");
+			}
+		} else {
+			assert!(false, "Expected variant");
+		}
+	} else {
+		assert!(false, "Expected named composite");
+	}
+
+	None
+}
```

Comment:
@jsdw could you please have a look at these functions? Is there a better way to do them?

Reply:
As discussed offline, this will probably easier with the static api. I'll give it a shot.

Reply:
Do let me know if you run into issues with that approach :)